src/OrchardCore.Modules/OrchardCore.Workflows/Services/WorkflowManager.cs

```diff
@@ -133,8 +134,12 @@ public async Task TriggerEventAsync(string name, IDictionary<string, object> inp
             // Start new workflows.
             foreach (var workflowType in workflowTypesToStart)
             {
+                // The 'TimerEvent' is a one minute signal and in fact this is the timer activity that checks if it has expired.
+                // So, because a timer event can't occur again before it has previously expired, if there is already a workflow
+                // instance that is halted on a timer activity, we should not start a new instance, we just have to resume it.
+
                 // If this is a singleton workflow and there's already an instance, then skip.
-                if (workflowType.IsSingleton && haltedWorkflows.Any(x => x.WorkflowTypeId == workflowType.WorkflowTypeId))
+                if ((workflowType.IsSingleton || name == TimerEvent.EventName) && haltedWorkflows.Any(x => x.WorkflowTypeId == workflowType.WorkflowTypeId))
```

Comment:
Fat hack. Maybe a custom property in tasks that would describe this behavior instead. Like what if another task needed to have the same behavior?

Reply:
Like `IsExclusive` or something similar

Reply:
Yes but here there are singularities that i tried to describe in my comments, one being that in fact the timer event is triggered every minute, one being that it is closely tied to the TimerEvent type ...

`IsExclusive` is a good idea, hmm i suggest a const property as it doesn't make sense to make it updatable

I will take a look.